tests/unit/src/util/Travis.ts

```diff
@@ -0,0 +1,264 @@
+import * as registerSuite from 'intern!object';
+import * as assert from 'intern/chai!assert';
+import loadModule, { cleanupModuleMocks } from '../../../_support/loadModule';
+import { stub, SinonStub } from 'sinon';
+import Travis, { Repository } from 'src/util/Travis';
+
+let module: any;
+let travis: Travis;
+let repository: Repository;
+let requestStub: SinonStub & Partial<{ get: SinonStub, post: SinonStub }>;
+
+registerSuite({
+	name: 'util/Travis',
+
+	before() {
+		requestStub = stub();
+		requestStub.post = stub();
+		requestStub.get = stub();
+	},
+
+	after() {
+		cleanupModuleMocks();
+	},
+
+	beforeEach() {
+		module = loadModule('src/util/Travis', {
+			'@dojo/core/request': { default: requestStub }
+		}, false);
+	},
+
+	afterEach() {
+		requestStub.reset();
+		requestStub.post.reset();
+		requestStub.get.reset();
+	},
+
+	'Travis': {
+		beforeEach() {
+			travis = new module.default();
+		},
+
+		async authenticate() {
+			const token = 'token';
+			const accessToken = 'access_token';
+			const post = requestStub.post;
+
+			post.returns(Promise.resolve({ json: () => Promise.resolve({ 'access_token': accessToken }) }));
+
+			const authenticate = await travis.authenticate(token);
+
+			assert.strictEqual(post.lastCall.args[1].body, '{"github_token":"token"}');
+			assert.strictEqual(travis.token, accessToken);
+			assert.strictEqual(authenticate, accessToken);
+		},
+
+		createAuthorization: (() => {
+			let repo: any;
+			const tokenAuth = { token: 'token' };
+
+			return {
+				beforeEach() {
+					repo = {
+						createAuthorization: stub().returns(Promise.resolve(tokenAuth)),
+						findAuthorization: stub().returns(Promise.resolve())
+					};
+				},
+
+				afterEach() {
+					repo.createAuthorization.reset();
+					repo.findAuthorization.reset();
+				},
+
+				async 'existing authorization; eventually throws'() {
+					repo.findAuthorization.returns(Promise.resolve({ id: 1 }));
+
+					try {
+						await travis.createAuthorization(repo);
+
+						assert.fail();
```

Comment:
Small nit-pick. The catch message will catch this and only fail on the error message assertion. We should explicitly fail it before it reaches the strict equal to make this test slightly better.